### DISCUSSION TITLE
how to make a pager in Razor Template for ListPart?

### DISCUSSION THREAD

**Discussion by hylinux** [OrchardCMS/OrchardCore#13669] [General]
Hi, I tried to search the document about this answer, but I failed to find it.
all of our document is talking about the liquid template how to make a pager.

if some one can show me a example for Razor Page template, it will help much.

also,   I kind of confuse "how to get a total number for item in a ListPart"?

thanks .

**Comment by hylinux** [comment]
Ok, look like I have to used Yesql get those item from DB and build the Pager

**Comment by hylinux** [comment]
finally, I got it. 

it is :

```csharp

@{

    var pager = Model.Pager;

    pager.PreviousText = "Previous Page";
    pager.NextText = "Next Page";
}
@await DisplayAsync(pager);
```

but the ListPart default use the `PagerSlim`, so, you can not custom more useful pager.  that's not good

**Reply by hishamco** [reply]
>  so, you can not custom more useful pager.

What do you mean?

**Reply by hylinux** [reply]
Hi, @hishamco  since ListPart used PagerSlim for Pager,   the page always just have two links: one is for next page, another one is for previous page,  if you want to add more links, like "first" and "end" or page with number, it doesn't work. have to re-coding.

**Reply by hishamco** [reply]
Frankly, I didn't try, you can check our docs, hope to find time to try it

**Reply by hylinux** [reply]
yep. I checked the document and the source code for ListPart, look like it is.
